File: backend/src/main/java/com/antigravity/module/permission/Permission.java
package com.antigravity.module.permission;

import com.antigravity.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;

/**
 * 权限实体
 *
 * @author Antigravity Team
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Permission extends BaseEntity {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 权限编码（唯一）
     */
    private String code;

    /**
     * 权限名称
     */
    private String name;

    /**
     * 权限类型：MENU-菜单, BUTTON-按钮, API-接口
     */
    private String type;

    /**
     * 父级ID
     */
    private Long parentId;

    /**
     * 路径/URL
     */
    private String path;

    /**
     * 图标
     */
    private String icon;

    /**
     * 排序号
     */
    private Integer sort;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 是否删除
     */
    private Boolean isDeleted;

}

File: frontend/src/types/permission.ts
/**
 * 权限类型枚举
 */
export enum PermissionType {
    MENU = 'MENU',
    BUTTON = 'BUTTON',
    API = 'API'
}

/**
 * 权限实体
 */
export interface Permission {
    id: number
    code: string
    name: string
    type: PermissionType
    parentId: number
    path?: string
    icon?: string
    sort: number
    enabled: boolean
    children?: Permission[]
    createTime: string
    updateTime: string
}

/**
 * 权限创建请求
 */
export interface PermissionCreateRequest {
    code: string
    name: string
    type: PermissionType
    parentId?: number
    path?: string
    icon?: string
    sort?: number
    enabled?: boolean
}

/**
 * 权限更新请求
 */
export interface PermissionUpdateRequest {
    name?: string
    parentId?: number
    path?: string
    icon?: string
    sort?: number
    enabled?: boolean
}

export const PermissionTypeMap: Record<string, string> = {
    [PermissionType.MENU]: '菜单',
    [PermissionType.BUTTON]: '按钮',
    [PermissionType.API]: '接口'
}

export const PermissionTypeColorMap: Record<string, string> = {
    [PermissionType.MENU]: 'blue',
    [PermissionType.BUTTON]: 'green',
    [PermissionType.API]: 'orange'
}
